os/fs/smartfs/smartfs_smart.c

```diff
@@ -747,6 +751,19 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 		goto errout_with_semaphore;
 	}
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
```

Comment:
I think added code in this function is not necessary.
in smartfs_sync_internal, we can add below of line 648, I think.
```
smartfs_setbuffer(&readwrite, sf->currsector, 0, fs->fs_llformat.availbytes, (uint8_t *)sf->buffer);
ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&readwrite);
if (ret < 0) {
	fdbg("Error %d writing used bytes for sector %d\n", ret, sf->currsector);
	goto errout;
}

sf->byteswritten = 0;
sf->bflags = 0;

/* If it is firstsector, update entry */
if (sf->entry->datlen == fs->fs_llformat.availbytes + sizeof(struct smartfs_chain_header_s)) {
      mode = entry->flag | .... // expand entry struct????.
       smartfs_writeentry(fs, sf->entry, sf->entry->name, SMARTFS_DIRENT_TYPE_FILE, mode, entry->dsector);
}
```

Reply:
Modified